### PR TITLE
ci: refactor fake status checks

### LINKED ISSUE #19289: CI: fake status checks produced even if Zeebe CI is running
## Description

The [operate-ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/operate-ci.yml) and [tasklist-ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/tasklist-ci.yml) GHA workflows report fake `Test summary` status checks under wrong conditions.
The `Test summary` status check is marked as `required` in GitHub and usually produced by [zeebe-ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/zeebe-ci.yml). In case that no change related to Zeebe CI happened (meaning it will not run), this status check is still required to allow merging PRs until https://github.com/camunda/camunda/issues/17721 is implemented which is why this workaround of faking (green) status checks exists.

Now we have the situation that due to overlapping path filters [operate-ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/operate-ci.yml) and [tasklist-ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/tasklist-ci.yml) report that fake status check (as green) _even if_ [zeebe-ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/zeebe-ci.yml) is running and before results are known. This can happen e.g. if a file in `parent` folder changes which triggers all mentioned GHA workflows.

### Impact

This can lead to PRs like https://github.com/camunda/camunda/pull/19223 with failing Zeebe tests being put into the merge queue due to the quicker faked (green) status check. The merge queue will reject this change later since it runs all tests so `main` will be clean. This merge queue failures can impact and stall other engineers' PRs.

#### Possible solutions

* adjust path filters to not overlap / prevent fake status check from being reported, also consider [common-ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/common-ci.yml)
* wait for https://github.com/camunda/camunda/issues/17721

### PR DESCRIPTION
**Pull request by cmur2**
## Description

The `Test summary` status check is marked as `required` in GitHub and usually produced by [zeebe-ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/zeebe-ci.yml). In case that no change related to Zeebe CI happened (meaning it will not run), this status check is still required to allow merging PRs until https://github.com/camunda/camunda/issues/17721 is implemented which is why this workaround of faking (green) status checks exists.

When only the [operate-ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/operate-ci.yml) existed it complemented exactly the `zeebe-ci.yml` since there were only those 2 cases. Later [tasklist-ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/tasklist-ci.yml) and the SDK got added and made the mix more complex.

Now we have the situation that due to overlapping path filters [operate-ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/operate-ci.yml) and [tasklist-ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/tasklist-ci.yml) report that fake status check (as green) _even if_ [zeebe-ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/zeebe-ci.yml) is running and before results are known. This can happen e.g. if a file in `parent` folder changes which triggers all mentioned GHA workflows.

This PR proposes to resolve that without impacting any other flow by removing the fake status checks completely from Operate CI and Tasklist CI (since they can't know when they run vs when Zeebe CI doesn't run).

Instead, the [common-ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/common-ci.yml) gets the exact opposite of the paths triggers (via ignore) of [zeebe-ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/zeebe-ci.yml). This means that if Zeebe CI is not producing the "Test summary" then the Common CI will run and fake it - only those 2 cases exist. Since neither Operate CI nor Tasklist CI are a required status check, they don't need to matter for this issue.

Mid-term this whole faking of status checks will become obsolete with #17721

## Related issues

Related to and should fix #19289

### CODE REVIEW

**Comment by lenaschoenburg** [comment]
> Since neither Operate CI nor Tasklist CI are a required status check, they don't need to matter for this issue.

But they are required? :confused: From https://github.com/camunda/camunda/settings/branch_protection_rules/2112808: 
image

File: .github/workflows/tasklist-ci.yml
Comment (houssain-barouni) [inline .github/workflows/tasklist-ci.yml:242]: Removing this will increase the likelihood of breaking changes introduced by Renovate in the Tasklist frontend.

Comment (cmur2) [reply .github/workflows/tasklist-ci.yml:242]: Are you sure?

To me this looks like a hack strapped upon our initial workaround of faking the "Test summary" status check of Zeebe CI. Comparing with https://github.com/camunda/camunda/blob/main/.github/workflows/operate-ci.yml#L85-L93 this fake "Test summary" job was initially without any dependencies or condition logic.

From Git history I can see that this logic was later added and if it is vital to the CI, it shouldn't be added to the same job that fakes the Zeebe CI for the required status check as that would be mixing multiple concerns...

**Comment by cmur2** [comment]
@lenaschoenburg sorry forgot to mention you above

File: .github/workflows/tasklist-ci.yml
Comment (houssain-barouni) [reply .github/workflows/tasklist-ci.yml:242]: Yes, unfortunately this is a workaround built on top of a workaround, to prevent Renovate breaking FE dependencies upgrades that were happening very frequently.

Comment (cmur2) [reply .github/workflows/tasklist-ci.yml:242]: Regarding the breaking FE dependency upgrades I'd like to understand if that would be still a frequent (returning) problem after merging this PR.

Looking [at the list of currently open Renovate PRs affecting Tasklist](https://github.com/camunda/camunda/pulls?q=is%3Aopen+is%3Apr+author%3Aapp%2Frenovate+label%3Acomponent%2Ftasklist) none of them seems like FE dependency upgrade. That means that all FE dependency upgrades were either already auto-merged (aka Tasklist CI incl FE tests was green) or there were not that much.

Does this make sense? Given the current status that is a risk I'm willing to take and worst case prioritize https://github.com/camunda/camunda/issues/17721 even more as a follow-up if the problem returns.

Comment (houssain-barouni) [reply .github/workflows/tasklist-ci.yml:242]: @cmur2 
> Looking [at the list of currently open Renovate PRs affecting Tasklist](https://github.com/camunda/camunda/pulls?q=is%3Aopen+is%3Apr+author%3Aapp%2Frenovate+label%3Acomponent%2Ftasklist) none of them seems like a FE dependency upgrade.

They were manually closed like this one https://github.com/camunda/camunda/pull/19235

>If Tasklist CI needs a status check we could either introduce a separate one like Zeebe's or wait until it is properly addressed in https://github.com/camunda/camunda/issues/17721

There is already a status check called "Tasklist CI test summary", but it is already returned by [tasklist-merge-ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/tasklist-merge-ci.yml). This one is also run in pull requests (for other reasons)
I think https://github.com/camunda/camunda/issues/17721 will be the answer to many of our problems

Comment (cmur2) [reply .github/workflows/tasklist-ci.yml:242]: @houssain-barouni are the Tasklist FE tests (or a relevant subpart of it) stable enough to be included in the https://github.com/camunda/camunda/blob/main/.github/workflows/tasklist-merge-ci.yml to prevent the dependency merges?

Comment (houssain-barouni) [reply .github/workflows/tasklist-ci.yml:242]: yes, I did not observe any flakiness related to these tests, @vsgoulart can you confirm ?

Comment (vsgoulart) [reply .github/workflows/tasklist-ci.yml:242]: > yes, I did not observe any flakiness related to these tests, @vsgoulart can you confirm ?

@houssain-barouni @cmur2 Yes, all frontend tests included in this CI are all stable

Comment (cmur2) [reply .github/workflows/tasklist-ci.yml:242]: I just discussed with @houssain-barouni how to move on here since I agree that merging breaking Renovate changes is not desirable. That said, the current hack used by the Tasklist CI to re-use the fake status check of the Zeebe CI to ensure that is also not acceptable.

A better solution is to extend https://github.com/camunda/camunda/blob/main/.github/workflows/tasklist-merge-ci.yml (which runs on every PR + is a required status check of its own) to run 1 of the Tasklist frontend CI jobs (e.g. `fe-tests`) given that the job is stable enough to not cause flaky builds.

The implementation of this solution needs be done by the HTO team since it requires knowledge about which Tasklist frontend CI job is representative to avoid broken Renovate merges + this task is related to exclusively the Tasklist CI.

Comment (cmur2) [reply .github/workflows/tasklist-ci.yml:242]: I created https://github.com/camunda/camunda/issues/19428 (blocking this PR) for the refactoring mentioned above and to be taken care of by the HTO team.

Comment (cmur2) [reply .github/workflows/tasklist-ci.yml]: @vsgoulart @houssain-barouni now that https://github.com/camunda/camunda/issues/19428 is resolved, are you fine with merging this?

**Review by houssain-barouni — APPROVED** [review]

**Pull request merged**
